### PR TITLE
Update bootstrap: Sample details and properties

### PR DESCRIPTION
**Pull request by maiwald**

### CODE REVIEW

File: app/packs/src/apps/mydb/elements/details/ElementDetails.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ import ReportContainer from 'src/apps/mydb/elements/details/reports/ReportContai
 import ResearchPlanDetails from 'src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails';
 import SampleDetails from 'src/apps/mydb/elements/details/samples/SampleDetails';
 import ScreenDetails from 'src/apps/mydb/elements/details/screens/ScreenDetails';
-import StickyDiv from 'react-stickydiv';
+import Sticky from 'react-sticky-el';
```
Comment (JoachimTillessen) [inline app/packs/src/apps/mydb/elements/details/ElementDetails.js:17]: what does it do exactly?

File: app/packs/src/apps/mydb/elements/details/samples/SampleDetails.js [hunk 2]
```diff
@@ -444,7 +451,7 @@ export default class SampleDetails extends React.Component {
 
     const saveAndCloseBtn = belongToReaction && !sample.isNew ? this.saveBtn(sample, true) : null;
     return (
-      <div className='d-flex gap-1'>
+      <div className="d-flex gap-1 mt-3">
```
Comment (JoachimTillessen) [inline app/packs/src/apps/mydb/elements/details/samples/SampleDetails.js:454]: This margin should be handled by tabs-container--with-borders. Let's checkout why it's missing together.

File: app/packs/src/apps/mydb/elements/details/samples/propertiesTab/ElementalComposition.js [hunk 3]
```diff
@@ -1,83 +1,56 @@
...
+      const value = data[key];
+      return (
+        <span key={key}>
+          <span className="fw-bold">
             {key}
-          </strong>
-          &nbsp;
-          &nbsp;
+          </span>
+          {' '}
```
Comment (JoachimTillessen) [inline app/packs/src/apps/mydb/elements/details/samples/propertiesTab/ElementalComposition.js:22]: Consider using spacing utility

File: app/packs/src/apps/mydb/elements/details/samples/propertiesTab/SampleDetailsSolventsDnd.js [hunk 4]
```diff
@@ -42,17 +42,17 @@ class SampleDetailsSolventsDnd extends React.Component {
...
-      style.borderColor = '#337ab7';
-    } else if (canDrop) {
-      style.borderStyle = 'dashed';
+
+    let className = '';
+    if (canDrop) {
+      className += ' border-dashed border-3';
+    }
+    if (canDrop && !isOver) {
+      className += ' border-gray-300';
```
Comment (JoachimTillessen) [inline app/packs/src/apps/mydb/elements/details/samples/propertiesTab/SampleDetailsSolventsDnd.js:51]: I possible use dedicated dnd utility classes from application.scss.erb

File: app/packs/src/apps/mydb/elements/details/samples/propertiesTab/SampleForm.js [hunk 5]
```diff
@@ -726,42 +659,46 @@ export default class SampleForm extends React.Component {
...
         {sample.isMethodDisabled('amount_value') ? (
-          <FormControl type="text" disabled defaultValue="***" readOnly />
+          <Form.Control type="text" disabled defaultValue="***" readOnly />
         ) : (
-          <div className='d-flex gap-2'>
+          <div className="d-flex gap-2">
             {this.numInput(sample, 'amount_g', 'g', ['m', 'n', 'u'], 4, null, 'massMgInput', isDisabled, '')}
             {this.numInput(sample, 'amount_l', 'l', ['m', 'u', 'n'], 5, null, 'l', isDisabled, '', volumeBlocked)}
             {this.numInput(sample, 'amount_mol', 'mol', ['m', 'n'], 4, null, 'amountInput', isDisabled, '')}
-            {sample.contains_residues && this.attachedAmountInput(sample)}
```
Comment (JoachimTillessen) [inline app/packs/src/apps/mydb/elements/details/samples/propertiesTab/SampleForm.js:735]: Is this correct?

File: app/packs/src/apps/mydb/elements/details/samples/propertiesTab/SampleForm.js [hunk 6]
```diff
@@ -726,42 +659,46 @@ export default class SampleForm extends React.Component {
...
-            variant={this.state.densityMolarity === 'density' ? 'info' : 'light'}
+            variant={densityMolarity === 'density' ? 'info' : 'light'}
             size="xxsm"
           >
             Density
           </Button>
           <Button
             onClick={() => this.setState({ densityMolarity: 'molarity' })}
-            variant={this.state.densityMolarity === 'molarity' ? 'info' : 'light'}
+            variant={densityMolarity === 'molarity' ? 'info' : 'light'}
```
Comment (JoachimTillessen) [inline app/packs/src/apps/mydb/elements/details/samples/propertiesTab/SampleForm.js:692]: We have dedicated Buttons for ButtonGroups with toggle functionality in another branch...

**Review by JoachimTillessen — CHANGES_REQUESTED** [review]
Looks good. Would like to ask a couple of questions, while looking at it together.

File: app/packs/src/apps/mydb/elements/details/samples/propertiesTab/ElementalComposition.js (hunk 3 above)
Comment (maiwald) [reply app/packs/src/apps/mydb/elements/details/samples/propertiesTab/ElementalComposition.js:22]: In this case, I think I prefer the actual space. If you copied the text, you'd get the key and value squished together, if we only space them using CSS.

**Pull request merged**